design/specs/specs-null-values.md

```diff
@@ -0,0 +1,176 @@
+4. The new value replaces the `NULL` value.
+
+#### Scenario 2b: Replacing `NULL` with an empty string
+
+##### Steps for 2b
+
+1. The user edits a `NULL` value in a column.
+2. The user enters `edit mode`.
+3. A text input box is displayed in `edit mode` with a placeholder set to `NULL`.
+4. The new typed-in value replaces the placeholder text.
+5. The user clears the new value by pressing the `backspace` key until the value is completely cleared.
+6. The user leaves `edit mode`.
+7. An empty string replaces the `NULL` value. The cell appears blank.
+
+#### Scenario 2c: The data type is boolean with checkbox display mode enabled
+
+##### Steps for 2c
+
+1. The user edits a `NULL` value in a column set to boolean type with the checkbox display mode enabled.
+    - The field contains a checkbox in an indeterminate state, meaning the value is `NULL`.
+2. The user clicks on the checkbox, going from indeterminate `NULL` to checked `TRUE`.
+3. The user can click on the checkbox again to mark it as `FALSE`.
+4. To return the value to `NULL`, the user can select the cell and press the `delete` key or use the contextual menu and select the `Set as NULL` option.
+
+#### Scenario 2d: The data type is number
+
+##### Steps for 2d
+
+1. The user edits a `NULL` value in a column set to `number` type.
+2. The user double-clicks the cell to enter `edit mode.
+3. The user leaves `edit mode` without making any changes.
+4. The `NULL` value remains unchanged.
+
+### Scenario 3: User sets a value to `NULL`
+
+#### Scenario 3a: The column accepts `NULL` values
+
+##### Steps for 3a
+
+1. The user selects a cell in a column that contains a value.
+2. The user clears the content of the selected cell.
+    - By pressing the `delete` key.
+    - By opening the contextual menu and selecting the `Set as NULL` option.
+3. The selected cell now has a `NULL` value.
+
+#### Scenario 3b: The column does not accept `NULL` values
+
+##### Steps for 3b
+
+1. The user selects a cell in a column that contains a value.
+2. The user attempts to clear the content of the selected cell.
+    - By pressing the `delete` key.
+        - The system prevents the insertion of a `NULL` value.
+        - An error message indicates that a NOT `NULL` constraint is applied to the column, and the system cannot insert `NULL` values.
+    - By opening the contextual menu and selecting the `Set as NULL` option.
+        - The 'Set as `NULL`' option is disabled, and the user cannot select it.
+
+## Interactions
+
+### Select vs. Edit Mode Behaviors
+
+In `Edit Mode`, a cell provides additional functionality and interactions to users. `Edit Mode` is toggled by double-clicking on a cell. However, certain data types could have different interactions. Once specific behaviors and components for each data type are introduced, we must further define these interactions to ensure compatibility.
```

Comment:
We decided to also enable `Edit Mode` if the user types a text character or backspace while in `Select Mode`.

Reply:
Would be good to also define `Select Mode` since it's referenced in the table below.